src/System.Management.Automation/engine/Utils.cs

```diff
@@ -1927,4 +1913,50 @@ internal T Pop()
             return item;
         }
     }
+
+    /// <summary>
+    /// A readonly Hashset.
+    /// </summary>
+    internal sealed class ReadOnlyHashSet<T> : IEnumerable
+    {
```

Comment:
You might consider using a more specialized name since this is no longer an `ISet`.

Reply:
I renamed the type to be `ReadOnlyBag`. That's the most appropriate name I could come up with. `ConcurrentBag` doesn't implement `ISet`, so I just borrowed `Bag` from it.